internal/csot/csot_test.go

```diff
@@ -0,0 +1,252 @@
+// Copyright (C) MongoDB, Inc. 2024-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package csot
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/internal/assert"
+)
+
+func newTestContext(t *testing.T, timeout time.Duration) context.Context {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	t.Cleanup(cancel)
+
+	return ctx
+}
+
+func newDurPtr(dur time.Duration) *time.Duration {
+	return &dur
+}
```

Comment:
Optional: Consider adding a generic version of this that works for any type, maybe to a package like `internal/testutil`.

E.g.
```go
func Ptr[T any](val T) *T {
	return &val
}
```
which you could use in a test like
```go
timeout: testutil.Ptr(time.Duration(2)),
```